XenServerHealthCheck/ServerListHelper.cs

```diff
@@ -280,7 +281,9 @@ public void UpdateXenCenterMetadata(string message)
                 string[] metadata = message.Split(SEPARATOR);
                 if (metadata.Length != 2) 
                     return;
-                Properties.Settings.Default.XenCenterMetadata = EncryptionUtils.UnprotectForLocalMachine(metadata[1]);
+                var metadataString = EncryptionUtils.UnprotectForLocalMachine(metadata[1]);
+                metadataString = metadataString.Replace(HealthCheckSettings.REPORT_TIME_PLACEHOLDER, DateTime.UtcNow.ToString("u"));

```

Comment:
This is wrong because the substitution happens at the wrong time. It needs to happen when (and each time) the report is sent to CIS, not when it is received from XenCenter.